### DISCUSSION TITLE
型違いの比較を行う方法について

### DISCUSSION THREAD

**Discussion by CC-Tani** [domaframework/doma#962] [Q&A] [answered]
Criteria APIを使った検索をかける際に、ge()を使用してPropertyMetamodel(LocalDate)とPropertyMetamodel(LocalDateTime)を比較したいと考えております。
同型のものの比較方法のみ把握している状況ですので、元となるメタデータのフィールドの型を変更して対応しています。
上記の方法でも一応処理は問題なく動作するのですが、よりよい方法はございますでしょうか。

**Comment by nakamura-to** [comment]
型が異なるプロパティの比較はサポートしていません。
型を合わせるような修正をしていただくのが良いのではと思います。

**Reply by CC-Tani** [reply]
DB項目
・日時項目A
・日付項目B

 対応するEntityクラス
・LocalDateTime型のA
・LocalDateTime型のB
・LocalDate型のB

※@Columnの指定は二つのBで同じものを指定している

Criteria APIでは「LocalDateTime型のA」と「LocalDateTime型のB」を用いて比較をしております。
上記の定義を行うことによる、懸念点などございますでしょうか。

**Reply by nakamura-to** [reply, marked as answer]
カラム1つに対し複数のプロパティをマッピングさせたいということでしょうか？
基本的には避けた方が良いと思います。
LocalDateとLocalDateTimeではより幅広いデータを表現できるのは後者なので、項目AもBもLocalDateTimeで表現するのが良いと思います（LocalDate型のプロパティ定義は不要だと思います）。
アプリケーションでLocalDate型が必要な時はLocalDateTime型から変換して利用すれば良いと思います。
